pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -390,14 +392,22 @@ def append_fn(v: ResultMetric) -> None:
         apply_to_collection(list(self.values()), ResultMetric, append_fn)
         return o
 
-    @property
-    def batch_size(self) -> torch.Tensor:
-        # performance: cache the `batch_size` tensor instead of re-creating it
-        return self._batch_size
-
-    @batch_size.setter
-    def batch_size(self, value: int) -> None:
-        self._batch_size = torch.tensor(value, device=self.device)
+    def _extract_batch_size(self, batch_size: Optional[int], meta: _Metadata) -> int:
+        # check if we have extracted the batch size already
+        batch_size = batch_size or self.batch_size
```

Comment:
@carmocca thanks for updates.. looks good.
this will fail with `self.log(..., batch_size=0)`. if the user is making a mistake, we shouldn't touch it.